### PR TITLE
Update to Laravel 5

### PR DESCRIPTION
**Pull request by Danno040**
### What is the problem / feature ?

Our sniffer rules command was simply not compatible with Laravel 5. This PR fixes that,

### How did it get fixed / implemented ?

- Updated `SniffCommand` to use a `passthru` to the code sniffer.
- Updated sniffs that relied on Laravel's Config, to use code sniffer's configuration instead.

### How can someone test / see it ?

- Add this branch to a repository where we use CS
- Run `php artisan sniff`

### (Possible) Additional Work

- [ ] Better encapsulate code sniffer configuration loading in sniffs
- [ ] Allow `SniffCommand` to accept command line options

### CODE REVIEW

File: src/Socialengine/SnifferRules/Command/SniffCommand.php [hunk 1]
```diff
@@ -232,4 +136,69 @@ protected function getOptions()
     {
         return array();
     }
+
+    protected function processOptions()
+    {
+        $standards = $this->config->get('sniffer-rules.standard', array('PSR2'));
```
Comment (smazurov) [inline src/Socialengine/SnifferRules/Command/SniffCommand.php]: lets use `[]`

File: src/Socialengine/SnifferRules/Command/SniffCommand.php [hunk 2]
```diff
@@ -232,4 +136,69 @@ protected function getOptions()
...
+        $files = $this->config->get('sniffer-rules.files', array('app/models', 'app/controllers'));
```
Comment (smazurov) [inline src/Socialengine/SnifferRules/Command/SniffCommand.php]: perhaps a bit semantic but we should probably use L5 standards here.

File: src/Socialengine/SnifferRules/Command/SniffCommand.php [hunk 3]
```diff
@@ -81,118 +60,43 @@ public function fire()
...
+        $this->info('Done.');
 
-        }
-        $options = array(
-            'standard' => $standard,
-            'files' => $files,
-            'ignored' => $ignored,
-            'extensions' => array('php'),
-        );
+        if ($exitCode !== 0) {
```
Comment (smazurov) [inline src/Socialengine/SnifferRules/Command/SniffCommand.php]: this stuff needs to fail on CI without user intervention.

File: src/Socialengine/SnifferRules/Command/SniffCommand.php (hunk 2 above)
Comment (Danno040) [inline src/Socialengine/SnifferRules/Command/SniffCommand.php]: You mean like [a Laravel specific standard](https://github.com/antonioribeiro/laravelcs)?

Comment (smazurov) [inline src/Socialengine/SnifferRules/Command/SniffCommand.php]: no, sorry, `app/models` and `app/controllers` [no longer](https://mattstauffer.co/blog/laravel-5.0-directory-structure-and-namespace) exist in laravel 5. I believe its safe to just do `app/`

Comment (Danno040) [inline src/Socialengine/SnifferRules/Command/SniffCommand.php]: Should I add `src/`?

Comment (smazurov) [inline src/Socialengine/SnifferRules/Command/SniffCommand.php]: no, because `src/` is not part of default L5 bootstrap.

**Comment by Danno040** [comment]
Anything else?

**Comment by smazurov** [comment]
no, assuming the checkboxes are the dreams that we can never accomplish

**Comment by Danno040** [comment]
I wouldn't saw we could **never** accomplish them, it's just not needed right now.

**Comment by smazurov** [comment]
:shipit:

**Pull request merged**